app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/SearchFragment.java

```diff
@@ -0,0 +1,204 @@
+
+        setBadgeDrawable(ResourcesCompat.getDrawable(context.getResources(),
+                R.drawable.ic_add_row_circle_black_24dp, context.getTheme()));
+
+        setTitle("Search Using Live Data");
+
+        setSearchResultProvider(this);
+
+        setOnItemViewClickedListener(new ItemViewClickedListener());
+
+        /**
+         * Set up the adapter which contains all the rows in current fragment page
+         */
+        mRowsAdapter = new ArrayObjectAdapter(new ListRowPresenter());
+        mRowsAdapter.clear();
+        mRelatedAdapter = new ListAdapter<>(new VideoCardPresenter());
+        HeaderItem header = new HeaderItem(0, "results");
+        mRowsAdapter.add(new ListRow(header, mRelatedAdapter));
+    }
+
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        mViewModel = ViewModelProviders.of(this).get(VideosViewModel.class);
+        subscribeUi(mViewModel);
+
+        NetworkLiveData.get(this.getActivity()).observe(this, new Observer<Boolean>() {
+            @Override
+            public void onChanged(@Nullable Boolean isNetworkAvailable) {
+                if (isNetworkAvailable) {
+                    getActivity().findViewById(R.id.no_internet_search).setVisibility(View.GONE);
+                } else {
+                    getActivity().findViewById(R.id.no_internet_search).setVisibility(View.VISIBLE);
+                }
+            }
+        });
+    }
+
+    @Override
+    public ObjectAdapter getResultsAdapter() {
+        return mRowsAdapter;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String newQuery) {
+
+        if (DEBUG) {
+            Log.i(TAG, String.format("Search Query Text Change %s", newQuery));
+        }
+
+        if (!TextUtils.isEmpty(newQuery) && !newQuery.equals("nil")) {
+            getActivity().findViewById(R.id.search_progressbar).setVisibility(View.VISIBLE);
+            newQuery = "%" + newQuery + "%";
+            mViewModel.setQueryMessage(newQuery);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+
+        if (DEBUG) {
+            Log.i(TAG, String.format("Search Query Text Submit %s", query));
+        }
+
+        if (!TextUtils.isEmpty(query) && !query.equals("nil")) {
+            getActivity().findViewById(R.id.search_progressbar).setVisibility(View.VISIBLE);
+            query = "%" + query + "%";
+            mViewModel.setQueryMessage(query);
+        }
+        return true;
+    }
+
+    private void subscribeUi(VideosViewModel viewModel) {
+        viewModel.getSearchResult().observe(this, new Observer<List<VideoEntity>>() {
+            @Override
+            public void onChanged(@Nullable List<VideoEntity> videoEntities) {
+
+                if (DEBUG) {
+                    Log.e(TAG, "onChanged: " + videoEntities );
+                }
+                if (videoEntities != null) {
```

Comment:
handle null as empty list? or does it never happen ?

Reply:
Done. The empty list and null has been handled.